src/Microsoft.TemplateEngine.Cli/Commands/BaseCommand.cs

```diff
@@ -90,13 +48,18 @@ protected IEngineEnvironmentSettings CreateEnvironmentSettings(GlobalArgs args,
             //for template instantiaton it has to be reparsed
             string? outputPath = ParseOutputOption(parseResult);
             IEngineEnvironmentSettings environmentSettings = new EngineEnvironmentSettings(
-                new CliTemplateEngineHost(_host, outputPath),
+                new CliTemplateEngineHost(_hostBuilder(parseResult), outputPath),
                 settingsLocation: args.DebugCustomSettingsLocation,
                 virtualizeSettings: args.DebugVirtualizeSettings,
                 environment: new CliEnvironment());
             return environmentSettings;
         }
 
+        protected ITelemetryLogger CreateTelemetryLogger(ParseResult parseResult)
```

Comment:
I can't see the whole picture, but seems like this could be private.

Reply:
It is protected due to it is used in `BaseCommand<T>`.